fastnml/reader.py

```diff
@@ -0,0 +1,255 @@
+from f90nml import Namelist, Parser
+import multiprocessing as mp
+import re
+from typing import List, Union
+
+_nml_types = Union[int, float, bool, str]
+_array_rg = re.compile('((?:[a-z][a-z0-9_]*))(\\()(\\d+)(\\))(.*)',
+                       re.IGNORECASE | re.DOTALL)
+
+###############################################################################
```

Comment:
These dividers are not PEP-8 compliant, which I know you don't care about, but I do.  Something like:
```
# -----------------------------------------
``` 
would be okay.